### PR TITLE
8287001: Add warning message when fail to load hsdis libraries

### PR DESCRIPTION
**Pull request by yuu1127**
When failing to load hsdis(Hot Spot Disassembler) library (because there is no library or hsdis.so is old and so on),
there is no warning message (only can see info level messages if put -Xlog:os=info).
This should show a warning message to tell the user that you failed to load libraries for hsdis.
So I put a warning message to notify this.

e.g.
`
/build/macosx-aarch64-server-fastdebug/jdk/bin/java -XX:+UnlockDiagnosticVMOptions -XX:+PrintAssembly -version
`
->
```
============================= C1-compiled nmethod ==============================
----------------------------------- Assembly ----------------------------------- 

[0.081s][warning][os] Try to load hsdis library failed
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8287001](https://bugs.openjdk.org/browse/JDK-8287001): Add warning message when fail to load hsdis libraries

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Yasumasa Suenaga](https://openjdk.org/census#ysuenaga) (@YaSuenag - **Reviewer**) ⚠️ Review applies to [58724253](https://git.openjdk.org/jdk/pull/8782/files/58724253db9397a14b97ad1ac9ab193e2e343690)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/8782/head:pull/8782` \
`$ git checkout pull/8782`

Update a local copy of the PR: \
`$ git checkout pull/8782` \
`$ git pull https://git.openjdk.org/jdk pull/8782/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8782`

View PR using the GUI difftool: \
`$ git pr show -t 8782`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/8782.diff">https://git.openjdk.org/jdk/pull/8782.diff</a>

</details>

### CODE REVIEW

**Review by vnkozlov — CHANGES_REQUESTED** [review]
Can you put the warning into `dll_load()`?
We already print messages there with `-XX:+Vebose` (unfortunately it is available only in debug VM).
Actually consider replacing print statements and `Verbose` check there with UL.

**Comment by yuu1127** [comment]
Hi @vnkozlov ,
I think if put warning into `Disassembler::dll_load()` , 
this would print duplicated error messages for each hsdis library as written in `Disassembler::load_library`.
```
  // Find the disassembler shared library.
  // Search for several paths derived from libjvm, in this order:
  // 1. <home>/lib/<vm>/libhsdis-<arch>.so  (for compatibility)
  // 2. <home>/lib/<vm>/hsdis-<arch>.so
  // 3. <home>/lib/hsdis-<arch>.so
  // 4. hsdis-<arch>.so  (using LD_LIBRARY_PATH)
```
Besides, message with `Verbose` check seems to check whether 
the length of the path of hsdis library is beyond the defined length ( not including the case of no library, or old version).

So, it seems reasonable to put the warning message into  `Disassembler::load_library` after finishing to check all patterns.

File: src/hotspot/share/compiler/disassembler.cpp [hunk 1]
```diff
@@ -837,6 +837,8 @@ bool Disassembler::load_library(outputStream* st) {
   if (_library != NULL) {
     _decode_instructions_virtual = CAST_TO_FN_PTR(Disassembler::decode_func_virtual,
                                           os::dll_lookup(_library, decode_instructions_virtual_name));
+  } else {
+    log_warning(os)("Try to load hsdis library failed");
```
Comment (JohnTortugo) [inline src/hotspot/share/compiler/disassembler.cpp:841]: NIT: I suggest adding the name of the file that was searched for and rewording this message a bit: "Failed to load hsdis library: hsdis-x86.so"

**Comment by JohnTortugo** [comment]
FWIW it seems to me that `load_library` is indeed the best place to print the message.

File: src/hotspot/share/compiler/disassembler.cpp (hunk 1 above)
Comment (yuu1127) [reply src/hotspot/share/compiler/disassembler.cpp]: @JohnTortugo 
Thank you for your advice !!
I added the name of the file to the warning message.

Comment (JohnTortugo) [reply src/hotspot/share/compiler/disassembler.cpp:841]: NIT: "Failed to load hsdis library." or "Loading hsdis library failed."

Comment (yuu1127) [reply src/hotspot/share/compiler/disassembler.cpp]: Thank you for your advice.
I changed the warning message to "Loading hsdis library failed".

**Comment by yuu1127** [comment]
PING: Could anyone review this?

**Review by YaSuenag — APPROVED** [review]
Looks good. I will sponsor you if you need.

**Comment by yuu1127** [comment]
@vnkozlov 
If you have some time, could you review again this?
As JohnTortugo also said, 
I believe `load library` is the best place to show the error message.

**Review by vnkozlov — APPROVED** [review]
Nice.

**Comment by yuu1127** [comment]
Thank you for reviewing @YaSuenag @vnkozlov  !!

**Pull request closed without merging**